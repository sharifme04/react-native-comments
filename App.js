/**
 * Soniq React Native App @ sharif hossain
 */

import React, {useEffect, useState} from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';
import axios from 'axios';
import {useDispatch, useSelector} from 'react-redux';

import ImageGalleryScreen from './app/screens/ImagesGallery/ImageGalleryScreen';
import {getImages} from './app/actions/image';
import {getComments} from './app/actions/comment';
import ModalComponent from './app/components/ModalComponent';

const App = () => {
  const dispatch = useDispatch();
  const [modalVisible, setModalVisible] = useState(false);
  const [modalImage, setModalImage] = useState(null);
  const [modalImageId, setModalImageId] = useState(null);
  const images = useSelector(state => state.imagesReducer?.images);
  const commentsReducer = useSelector(state => state.commentsReducer);

  //console.log('comments' ,commentsReducer?.comments); 

  useEffect(() => {
    dispatch(getImages());
    dispatch(getComments())
  }, []);

  const handleVisible = visible => setModalVisible(visible);
  const handleImage = (image, id) => {
    setModalImage(image);
    setModalImageId(id);
  };

  return (
    <SafeAreaView style={styles.container}>
      <ModalComponent
        handleVisible={handleVisible}
        modalImage={modalImage}
        modalVisible={modalVisible}
        id={modalImageId}
      />
      <ImageGalleryScreen
        images={images?.photos}
        handleVisible={handleVisible}
        handleImage={handleImage}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});

export default App;
