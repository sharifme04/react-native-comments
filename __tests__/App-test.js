/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import App from '../App';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const mockStore = configureMockStore();
const store = mockStore({});

it('renders correctly', () => {
  renderer.create(
    <Provider store={store}>
      <App />
    </Provider>,
  );
});


test('renderer snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
