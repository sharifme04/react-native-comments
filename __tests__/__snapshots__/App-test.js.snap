// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    propagateSwipe={true}
    scrollOffset={1}
    scrollTo={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.7)",
        "padding": 40,
      }
    }
    transparent={true}
    visible={false}
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.7)",
              "padding": 40,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "borderColor": "#fff",
                "color": "#fff",
                "fontSize": 15,
                "fontWeight": "bold",
              }
            }
          >
            Close
          </Text>
          <Image
            source={
              Object {
                "uri": null,
              }
            }
            style={
              Object {
                "alignSelf": "stretch",
                "height": 800,
                "width": null,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginLeft": 9,
                "marginTop": -60,
                "minHeight": 40,
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Comment here ..."
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#fff",
                  "borderWidth": 1,
                  "marginHorizontal": 0,
                  "padding": 10,
                  "width": "70%",
                }
              }
              value=""
            />
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Post
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </Modal>
  <View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      style={
        Object {
          "backgroundColor": "#eee",
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
